src/test/java/tests/SearchTests.java

```diff
@@ -0,0 +1,93 @@
+    }
+
+    @Test(priority = 3, description = "JIRA Case- 80982. Test for Searching artist 'Grav'.", dataProvider = "searchArtistData", dataProviderClass = DataProviders.class)
+    public void searchForArtistGrav (String email, String password, String artist){
+        loginPage = new LoginPage(driver);
+        searchPage = new SearchPage(driver);
+
+        loginPage.userLogin(email, password)
+                .verifyLoginSuccessful();
+        searchPage.searchArtistGrav(artist)
+                .verifyGravResultsMatchesArtistSectionOnResultsPage();
+    }
+
+    @Test(priority = 4, description = "JIRA Case- 80984. Test for Searching song/artist/album that doesn't exist.", dataProvider = "searchNonExistingSongNameData", dataProviderClass = DataProviders.class)
+    public void searchNonExistingSong (String email, String password, String Song) {
+        loginPage = new LoginPage(driver);
+        searchPage = new SearchPage(driver);
+
+        loginPage.userLogin(email, password)
+                .verifyLoginSuccessful();
+        searchPage.searchNonExistingSong(Song)
+                .verifyResultIsAnEmptyList();
+    }
+
+    @Test(priority = 5, description = "JIRA Case- 80985 and 80986. Test for Search should ignore trailing  white spaces.", dataProvider = "searchWhiteSpacesData", dataProviderClass = DataProviders.class)
+    public void searchIgnoresWhiteSpaces (String email, String password, String Song){
+        loginPage = new LoginPage(driver);
+        searchPage = new SearchPage(driver);
+
+        loginPage.userLogin(email, password)
+                .verifyLoginSuccessful();
+        searchPage.searchSongPluto(Song)
+                .verifyPlutoResultsMatchesSongSectionOnResultsPage();
+    }
+
+    @Test(priority = 6, description = "JIRA Case- 80987. Test for Search should be case sensitive", dataProvider = "searchCaseSensitiveData", dataProviderClass = DataProviders.class)
+    public void searchShouldBeCaseSensitive (String email, String password, String Song) {
+        loginPage = new LoginPage(driver);
+        searchPage = new SearchPage(driver);
+
+        loginPage.userLogin(email, password)
+                .verifyLoginSuccessful();
+        searchPage.searchSongPluto(Song)
+                .verifyWhenEnteringCaseSensitiveDataShouldNotBeValid();
+    }
+
+    @Test(priority = 7, description = "JIRA Case- 80988. Test User can clear the search query with keyboard esc key and 'x' button", dataProvider = "searchPressingESCKeyData", dataProviderClass = DataProviders.class)
+    public void searchClearingSearchBar (String email, String password, String Song) throws InterruptedException {
+        loginPage = new LoginPage(driver);
+        searchPage = new SearchPage(driver);
+
+        loginPage.userLogin(email, password)
+                .verifyLoginSuccessful();
+        searchPage.searchSongPluto(Song);
+        searchPage.verifyPlutoResultsMatchesSongSectionOnResultsPage();
+        Thread.sleep(1000);
```

Comment:
Please, delete `throws InterruptedException`

Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.